pulldown-cmark/src/strings.rs

```diff
@@ -260,6 +260,10 @@ impl<'a> CowStr<'a> {
             CowStr::Inlined(s) => s.deref().to_owned(),
         }
     }
+
+    pub fn into_static(self) -> CowStr<'static> {
+        self.into_string().into()
```

Comment:
I think this should take advantage of the fact that only the `Borrowed` variant needs to reallocate, others can be no-ops.

Reply:
Thank you.

The variants looked a bit complicated so I went an easy route previously. Fixed it now.

Reply:
One more improvement would be to utilize the Inline variant if possible. The Clone implementation has an example of doing this.

```rust
match InlineStr::try_from(s) {
    Ok(inline) => CowStr::Inlined(inline),
    Err(..) => CowStr::Boxed(s.into()),
},
```

Reply:
I don't need to do that for the `Boxed` variant as well, right? Since, we're consuming the `self` and passing current `Box<str>` to output.

Reply:
Correct. Only the Borrowed variant needs to be changed in this case. Cloning is a bit different.